sympy/solvers/tests/test_inequalities.py

```diff
@@ -389,7 +389,7 @@ def test_issue_10198():
 def test_issue_10047():
     # this must remain an inequality, not True, since if x
     # is not real the inequality is invalid
-    assert solve(sin(x) < 2) == (x <= oo)
+    assert solve(sin(x) < 2) == True
 
```

Comment:
I changed this test because, somewhere in the execution of `solve`, `reduce_inequalities` is called on `sin(x) < 2` which adds the assumptions `extended_real = True` on `x`. The result therefore changes, since with this PR ` x <= oo ` for `extended_real`.

Reply:
Is this fixable? It could be punted to a new issue if it is difficult...

Reply:
I can't argue what `solve(sin(x) < 2)` should return. Apparently what `solveset` does in the same scenario is that it rejects such queries where the arguments are not real. I believe this should be the case. The whole idea of enforcing `extended_real` assumptions into arguments does not seem valid to me. But if pushing assumptions is required, the results I am getting is valid. From `solve`'s
perspective `S.Reals` should be a better result. But I don't completely understand `solve`'s API.

Reply:
I think that `solve` has a poorly designed API. Too many different possibilities are crammed into one function. The inequality solving in particular is misleading and should really be in a separate function.

If the inequality solver is based on the assumption that unknowns are real then the result here should clearly be True.

Reply:
The comment should be updated if this is changed. The comment should describe what happened before and what is changed here